types/block_test.go

```diff
@@ -113,6 +113,44 @@ func TestBlockID(t *testing.T) {
 	}
 }
 
+// TestHeaderID probes the ID function of the BlockHeader type.
+func TestHeaderID(t *testing.T) {
```

Comment:
might make sense to throw a sanity check somewhere in the consensus code to make sure that the block id and the header id always yield the same result. Then we'd get to verify it when running a debugging instance of 'siad', because it would check across a wide range of block types.